### PR TITLE
cli: brocfile path option

### PR DESCRIPTION
**Pull request by chrmod**
implements part of #331

### CODE REVIEW

**Comment by chrmod** [comment]
@stefanpenner I'm happy to implement the rest of #331

File: lib/load_brocfile.js [hunk 1]
```diff
@@ -4,10 +4,16 @@ const path = require('path');
 const findup = require('findup-sync');
 
 module.exports = loadBrocfile;
-function loadBrocfile() {
-  const brocfile = findup('Brocfile.js', {
-    nocase: true,
-  });
+function loadBrocfile(brocfilePath) {
```
Comment (stefanpenner) [inline lib/load_brocfile.js]: lets add some unit tests to this function to ensure we don't regress

**Comment by stefanpenner** [comment]
> @stefanpenner I'm happy to implement the rest of #331

I would love that!

File: test/load_brocfile_test.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      chai.expect(loadBrocfile()).to.equal(brocfileFixture);
+    });
+  });
+
+  context('with Brocfile.js, called in subfolder', function() {
+    beforeEach(function() {
+      process.chdir(projectPath + '/subdir');
+    });
+
+    afterEach(function() {
```
Comment (stefanpenner) [inline test/load_brocfile_test.js]: i dont think this `afterEach` is required, as the parent describe blocks afterEach already restores the `chdir`

**Comment by chrmod** [comment]
@stefanpenner alright. Guess more tests for `load_brocfile` will arrive with `cwd` option.

File: test/load_brocfile_test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    });
+
+    it('return tree definition', function() {
+      chai.expect(loadBrocfile()).to.equal(brocfileFixture);
+    });
+  });
+
+  context('with path', function() {
+    it('return tree definition', function() {
+      chai.expect(loadBrocfile(projectPath + '/Brocfile.js')).to.equal(brocfileFixture);
```
Comment (stefanpenner) [inline test/load_brocfile_test.js]: can we also test the error case here? Specifically that if an explicit path is given, but that path doesn't contain a file that we hit the helpful error?

File: test/load_brocfile_test.js (hunk 2 above)
Comment (chrmod) [reply test/load_brocfile_test.js]: it is needed as `load_brocfile` changes working dir https://github.com/broccolijs/broccoli/blob/master/lib/load_brocfile.js#L18

File: test/load_brocfile_test.js (hunk 3 above)
Comment (chrmod) [reply test/load_brocfile_test.js]: make sense :+1:

File: test/load_brocfile_test.js (hunk 2 above)
Comment (stefanpenner) [reply test/load_brocfile_test.js]: but doesn't the outer `afterEach` already `chdir`? Am I confused and that one doesn't fire?

File: test/load_brocfile_test.js (hunk 3 above)
Comment (stefanpenner) [reply test/load_brocfile_test.js]: thx

File: test/load_brocfile_test.js (hunk 2 above)
Comment (chrmod) [reply test/load_brocfile_test.js]: right, sorry, it's pretty late here

File: test/load_brocfile_test.js (hunk 3 above)
Comment (chrmod) [reply test/load_brocfile_test.js]: done

File: test/load_brocfile_test.js (hunk 2 above)
Comment (stefanpenner) [reply test/load_brocfile_test.js]: no worries

**Comment by stefanpenner** [comment]
LGTM

**Comment by stefanpenner** [comment]
rebasing

**Comment by stefanpenner** [comment]
rebased -> https://github.com/broccolijs/broccoli/pull/339

**Pull request closed without merging**